terraform/modules/data-pipelines/avl-pipeline/main.tf

```diff
@@ -467,3 +455,279 @@ resource "aws_ecs_service" "siri_vm_generator_service" {
     ignore_changes = [task_definition]
   }
 }
+
+resource "aws_iam_policy" "siri_vm_downloader_ecs_execution_policy" {
+  count = var.environment != "local" ? 1 : 0
+
+  name = "integrated-data-siri-vm-downloader-ecs-execution-policy-${var.environment}"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Action" : "logs:CreateLogGroup",
+        "Resource" : [
+          "arn:aws:logs:eu-west-2:${data.aws_caller_identity.current.account_id}:log-group:/ecs/siri-vm-downloader-${var.environment}:*"
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role" "siri_vm_downloader_ecs_execution_role" {
+  count = var.environment != "local" ? 1 : 0
+
+  name = "integrated-data-siri-vm-downloader-ecs-execution-role-${var.environment}"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "ecs-tasks.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy",
+    aws_iam_policy.siri_vm_downloader_ecs_execution_policy[0].arn
+  ]
+}
+
+resource "aws_iam_policy" "siri_vm_downloader_ecs_task_policy" {
+  count = var.environment != "local" ? 1 : 0
+
+  name = "integrated-data-siri-vm-downloader-ecs-task-policy-${var.environment}"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Action" : "s3:GetObject",
+        "Resource" : [
+          "${aws_s3_bucket.integrated_data_avl_siri_vm_bucket.arn}/*"
+        ]
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : "secretsmanager:GetSecretValue",
+        "Resource" : [
+          var.db_secret_arn
+        ]
+      },
```

Comment:
```suggestion
```

If we're not fetching the api key in the ECS task we shouldn't need this permission any more :)

Reply:
It needs it to retrieve the db creds